### ISSUE TITLE
solidity-coverage hardhat plugin override the network object

### ISSUE THREAD

**Issue by wighawag** [sc-forks/solidity-coverage#624]
This line override the hre.network object : https://github.com/sc-forks/solidity-coverage/blob/3c0f3a5c7db26e82974873bbf61cf462072a7c6d/plugins/resources/nomiclabs.utils.js#L136-L141

This conflict with plugins that add field to the network object. (an example of such plugin is `hardhat-deploy`)

solidity-coverage should keep existing configuration.

**Comment by cgewecke** [comment]
Yes definitely, will fix.

**Comment by cgewecke** [comment]
@wighawag From your perspective, is the correct way to do this to merge with the existing properties on `network`? 

e.g
```js
env.network = Object.assign({ 
   name: networkName, 
   config: networkConfig, 
   provider: provider, 
   isHardhatEVM: isHardhatEVM 
 }, env.network) 
```

Or is this not enough? What modifications are you making on your end?

**Comment by wighawag** [comment]
Yes that would work. I am adding extra field to the network object that deploy scripts can then used to perform different things depending on their values: {tags, live, saveDeployments}.

But there was actually a bigger issue for hardhat-deploy.
In a recent version I was accessing hre.network.provider at import time and storing it in a variable.
This caused hardhat-deploy to hang on that version, ignoring the fact that solidity-coverage replaced with a new one on task execution.

This is I think a general problem with hardhat plugin api where it is not clear for plugin at what point they get called in relation to other plugins, hence why I created : https://github.com/nomiclabs/hardhat/issues/1374

I fixed it in hardhat-deploy by storing a lazyObject to the provider. This allow solidity-coverage to replace it and hadhat-deploy to get the modified version when it is actually needed
